lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1520,7 +1520,11 @@ namespace Js
         ScriptContext* scriptContext = instance->GetScriptContext();
         DynamicType* cachedDynamicType = nullptr;
         BOOL isJsrtExternalType = instance->GetType()->IsJsrtExternal();
-        DynamicType* oldType = isJsrtExternalType ? 0 : instance->GetDynamicType();
+        // Share the type of all JsrtExternal objects which will have 'newPrototype' as their new prototype.
+        // TypeTransitionMap is a map with key as weakReference and to store it, key need to have a valid heap block.
+        // So instead of using some constant, use newPrototype as the key for saving newType for such objects
+        // in TypeTransitionMap.
+        DynamicType* oldType = isJsrtExternalType ? reinterpret_cast<DynamicType*>(newPrototype) : instance->GetDynamicType();
```

Comment:
I don't understand this bit but the rest of the change is fine. Why can't you use newPrototype->GetType() here to use a real type instead of pretending like an object is a type- seems dangerous. But I'll defer to @curtisman/@pleath here